File: Front-end/src/types/image.ts
export default class Image
{
  private _path: string;
  private _height: number;
  private _width: number;

  constructor(path: string, width: number, height: number)
  {
    this._path = path;
    this._height = height;
    this._width = width;
  }
  
  public get path(): string { return this._path }
  public get height(): number { return this._height }
  public get width(): number { return this._width }
}

File: Back-end/src/main/java/com/springNAS/Back_end/Types/Image.java
package com.springNAS.Back_end.Types;

import java.security.InvalidParameterException;

public record Image(String path, int width, int height) {
  public Image(String path, int width, int height)
  {
    this.path = path;
    this.width = width;
    this.height = height;

    checkInvariants();
  }

  private void checkInvariants()
  {
    if(this.path == null) throw new InvalidParameterException("Image path must be non-null.");
    if(this.width <= 0) throw new InvalidParameterException("Image width must be positive.");
    if(this.height <= 0) throw new InvalidParameterException("Image height must be positive.");
  }
}
